dotnet-desktop-guide/net/winforms/printing/overview.md

```diff
@@ -0,0 +1,154 @@
+  
+### To change print options programmatically  
+  
+01. Add a <xref:System.Windows.Forms.PrintDialog> and a <xref:System.Drawing.Printing.PrintDocument> component to your form.  
+  
+01. Set the <xref:System.Windows.Forms.PrintDialog.Document%2A> property of the <xref:System.Windows.Forms.PrintDialog> to the <xref:System.Drawing.Printing.PrintDocument> added to the form.  
+  
+    ```vb  
+    PrintDialog1.Document = PrintDocument1  
+    ```  
+  
+    ```csharp  
+    printDialog1.Document = PrintDocument1;  
+    ```  
+  
+01. Display the <xref:System.Windows.Forms.PrintDialog> component by using the <xref:System.Windows.Forms.CommonDialog.ShowDialog%2A> method.  
+  
+    ```vb  
+    PrintDialog1.ShowDialog()  
+    ```  
+  
+    ```csharp  
+    printDialog1.ShowDialog();  
+    ```  
+  
+01. The user's printing choices from the dialog will be copied to the <xref:System.Drawing.Printing.PrinterSettings> property of the <xref:System.Drawing.Printing.PrintDocument> component.  
+
+## How to: Create Standard Windows Forms Print Jobs
+
+The foundation of printing in Windows Forms is the <xref:System.Drawing.Printing.PrintDocument> component—more specifically, the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event. By writing code to handle the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event, you can specify what to print and how to print it.  
+  
+### To create a print job  
+  
+01. Add a <xref:System.Drawing.Printing.PrintDocument> component to your form.  
+  
+01. Write code to handle the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event.  
+  
+     You'll have to code your own printing logic. Additionally, you'll have to specify the material to be printed.  
+  
+     As a material to print, in the following code example, a sample graphic in the shape of a red rectangle is created in the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event handler.  
+  
+    ```vb  
+    Private Sub PrintDocument1_PrintPage(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintPageEventArgs) Handles PrintDocument1.PrintPage  
+       e.Graphics.FillRectangle(Brushes.Red, New Rectangle(500, 500, 500, 500))  
+    End Sub  
+    ```  
+  
+    ```csharp  
+    private void printDocument1_PrintPage(object sender,
+    System.Drawing.Printing.PrintPageEventArgs e)  
+    {  
+       e.Graphics.FillRectangle(Brushes.Red,
+         new Rectangle(500, 500, 500, 500));  
+    }  
+    ```  
+  
+     (Visual C#) Place the following code in the form's constructor to register the event handler.  
+  
+    ```csharp  
+    this.printDocument1.PrintPage += new  
+       System.Drawing.Printing.PrintPageEventHandler  
+       (this.printDocument1_PrintPage);  
+    ```  
```

Comment:
This odd double snippet to demonstrate one thing causes issues when someone has the other language selected when browsing the article. This is where you could use multiple ranges to present a single code snippet. The range would include a code comment that handled the text above

The range would be like `:::code language="csharp" source="snippets/...file..." range="47-67,100-103":::`

```csharp
// This code would be in the constructor or set via the designer
this.printDocument1.PrintPage += new  
       System.Drawing.Printing.PrintPageEventHandler(this.printDocument1_PrintPage);  

// Event handler
private void printDocument1_PrintPage(object sender, System.Drawing.Printing.PrintPageEventArgs e)  
{  
    e.Graphics.FillRectangle(Brushes.Red, new Rectangle(500, 500, 500, 500));  
}  
```

Reply:
Got it. But source of this document does not have code snippets, so adding inline code block. The above-mentioned style is used for second article.

Reply:
@rachana-satao Correct, so a new snippets project would have to be created and these snippets migrated. Do you think you can do this? If not, I can file an issue for myself to track and I'll do the migrations at a later date.

Reply:
Yes, I am done with the csharp code. Sending you the sample project by email.